### ISSUE TITLE
this.state.viewport is undefined

### ISSUE THREAD

**Issue by sweeped** [root-two/react-native-drawer#140]
From package.json:

``` json
` "dependencies": {
    "babel-core": "^6.7.6",
    "babel-preset-react-native": "^1.5.6",
    "babel-relay-plugin": "^0.8.0",
    "react": "^0.14.5",
    "react-native": "^0.23.1",
    "react-native-drawer": "2.0.0"
  }
```

And using components copied from the example project (Main and ControlPanel), upon running the following error message displays:

```
undefined is not an object (evaluating '_this.state.viewport')
```

Has anyone else come across this issue? Is this indeed a bug or have I missed something?

Thanks!

**Comment by rt2zz** [comment]
Odd, state is defined as a class property on the drawer, possibly something changed with the way class properties work in rn 0.23?i will have to test this out

**Comment by sweeped** [comment]
I tested out the package again, following the example's package.json, and installing the packages manually, one by one (including react-native 0.24.1), and it worked as fine. Also I added several other packages manually, and it continued to work.

Later, however, after attempting to install all packages with `npm install` the error returns. Any ideas why it works when installing packages individually, in a particular order, but fails on a general install?

**Comment by rt2zz** [comment]
I have no idea, and cannot replicate :(

If you can update to the latest that I just published `npm i --save react-native-drawer@next`.

**Comment by cdlewis** [comment]
Seeing this on 0.25.1, downgrading to 0.24.1 fixes.

**Comment by rt2zz** [comment]
fwiw I have had no problems on 0.24 or 0.25. That said I am open to workarounds, possibly moving setting initial state to a constructor.

**Comment by cdlewis** [comment]
Yeah you're right, I don't know what I did the first time but it clearly wasn't downgrading that fixed it ><

I am pretty sure that going back to 1.16.7 did though. I think the issue is that ES6 classes don't support autobinding.

So you used to manually bind your methods: https://github.com/root-two/react-native-drawer/blob/bf49833f28e584c0b3fe8a6b796f2d3ed19a9ad5/index.js#L94

**Comment by keokilee** [comment]
I've just encountered this error and it's not working for me either. It looks like `this` isn't being set properly when an auto-bound method is used. I'm using`babel-preset-react-native@^1.6.0`and `react-native@^0.29.1` if that makes a difference.

**Comment by keokilee** [comment]
Looks like the fix for me was to install `babel-plugin-transform-class-properties` and add the following to my `.babelrc`:

```
  "plugins": [
    "transform-class-properties"
  ]
```

**Comment by rt2zz** [comment]
@keokilee glad you got it working. react-native babel plugin comes with class properties by default: https://github.com/facebook/react-native/blob/master/babel-preset/plugins.js#L16

Not sure why you were having issues. One option to avoid this confusion in the future is to babelify our source before publishing to npm. I am open to community arguments on the matter, although so far it seems most react-native projects do not compile src.

**Comment by janaka** [comment]
Same issue. None of the above worked. I do have it running properly is a differnet project with the same dependencies.

```
  "dependencies": {
    "babel-preset-react-native": "1.9.0",
    "babel-relay-plugin": "^0.9.3",
    "fbemitter": "^2.0.2",
    "react": "15.3.1",
    "react-native": "0.32.0",
    "react-native-drawer": "^2.3.0",
    "react-native-material-kit": "^0.3.2",
    "react-native-vector-icons": "^2.0.3",
    "react-relay": "^0.9.3",
    "underscore": "^1.8.3"
  },
  "devDependencies": {
    "babel-cli": "^6.14.0",
    "babel-core": "6.9.1"
  },
  "engines": {
    "npm": ">=3"
  }
```

**Comment by ssomnoremac** [comment]
I'm having this same issue @rt2zz . Fixed mine with #216 advice to change `.babelrc`

```
"presets": [
      -   "react-native"
      +   "react-native-stage-0"
  ]
```
